### ISSUE TITLE
Why the last part of the path to Julia binary is hardcoded?

### ISSUE THREAD

**Issue by eugenio** [forio/julia-studio#190]
Error: Failed to start Julia.
Expected location: /opt/Julia_Studio/bin/julia-basic/bin/julia-basic

Seems to me that the last part of the path to julia binary is hardcoded, is it meant to be like this or is just a mistake?

**Comment by stuinzuri** [comment]
There are two Julia binaries provided in a distribution, and Julia Studio symbolically links to the julia-basic version.  The path /opt/Julia_Studio/bin/julia-basic/bin/julia-basic looks strange to me.  

Can you please explain how you started Julia Studio?

As a side note, we are working on distributing Julia Studio via a PPA which should make these problems less likely.

**Comment by WestleyArgentum** [comment]
The point about how you specify the Julia executable is fair enough though, we should be able to run Julia regardless of the executable name (it doesn't work that way now because of historical compatibility issues on windows - but those are long gone).

We should update the "Path To Julia" setting to be a full path to an executable and not make assumptions.

**Comment by stuinzuri** [comment]
How would it behave is someone pointed to the 'wrong' julia executable?  I understand that the difference between the two is readline support.

**Comment by WestleyArgentum** [comment]
True, it would print things strangely. Maybe that's something to handle with a message above the "Path To Julia" field?

Actually, though, we should really just do some validation on their selection (rather than making them save and go back to the console to see if they did it right). So maybe we do a quick check in the background when they modify the path to make sure they pointed at a proper Julia binary?

**Comment by stuinzuri** [comment]
Part of the reason this came up was that the initial distro i built had an incorrect sym link.  (`julia` instead of `julia-basic`.)  Since the Linux distro doesn't include Julia, and shouldn't, let's change the default binary location to `/usr/bin/julia-basic`.  That removes the symlink issue and probably 99% of problems related to "where in the world julia binary?"

**Comment by keesvp** [comment]
For Windows, this issue is taken over by #204

**Comment by CNOT** [comment]
I am using Ubuntu 14.04 and I still have this problem. I tried to install Julia Studio using:
https://github.com/forio/julia-studio/blob/master/linux-build.md

**Comment by stochasticTarget** [comment]
I have the same problem as CNOT but following http://academic-onion.org/2014/03/25/installing-julia-studio-on-ubuntu-13-10/comment-page-1/#comment-114

**Comment by CNOT** [comment]
The error I was getting recently was:

Error: Failed to start Julia.
Expected location: /julia-basic

I managed to resolve that by looking into the /usr/bin directory for the julia application file. The name of the file in my case was just `julia`. So I tried this command line:

sudo ln -s /usr/bin/julia /julia-basic

Now when I start JuliaStudio I get a different error:

QProcessPrivate::execChild() failed to chdir to /share/julia-studio/Console
ERROR: could not open file /share/julia-studio/Console/Console.jl
 in include at ./boot.jl:245
 in include_from_node1 at loading.jl:128
 in process_options at ./client.jl:285
 in _start at ./client.jl:354
 in _start_3B_1713 at /usr/bin/../lib/x86_64-linux-gnu/julia/sys.so

**Comment by ibadami** [comment]
Did you resolved your issue CNOT?
I guess the problem is, after make install console folder is not created in julia-studio. 
Does anyone else face the same problem?

**Comment by CNOT** [comment]
No. Unfortunately I couldn't make Julia Studio work. I ended up using julia in sublime text, which has a really nice interface but nevertheless it's lacking some of the graphical aspects of Julia Studio.

**Comment by ibadami** [comment]
Thank you for the reply.

On Mon, Jun 15, 2015 at 8:14 PM, CNOT <notifications@github.com> wrote:

> No. Unfortunately I couldn't make Julia Studio work. I ended up using
> julia in sublime text, which has a really nice interface but nevertheless
> it's lacking some of the graphical aspects of Julia Studio.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/forio/julia-studio/issues/190#issuecomment-112159307>.
>



-- 

Ishrat Badami
Ph.D. Student
Computer Vision Group
UMIC Research Centre
RWTH Aachen University
Room No 129
Mies-van-der-Rohe Strasse 15
D-52074 Aachen
Germany
------------------------------------
Tel: +49 241 80 20775
Fax: +49 241 80 22731

e-mail: badami@vision.rwth-aachen.de
URL: http://www.vision.rwth-aachen.de/people/ishrat-badami
-----------------------------------------------------------------

**Comment by ibadami** [comment]
Hi CNOT,

I just figured out how to fix the following error.
QProcessPrivate::execChild() failed to chdir to /share/julia-studio/Console
ERROR: could not open file /share/julia-studio/Console/Console.jl

When I checked in the /share/julia-studio/  path there was no Console
folder present.
I looked at the output of make install step for julia-studio, it copied the
share folder of julia-studio in /share/qtcreator instead of
/share/julia-studio.
When I copied it in the right location '/share/julia-studio/' and run the
julia-studio.sh file, it worked without any problem.
I wrote a small code in the console and it give the right result.
If you wish you can try to see if this is the case for you too.
I have installed julia release-0.3 and julia-0.3-compatibility in my home
directory and set LD_LIBRARY_PATH in ~/.profile file. My qt version is 5.4
which is also installed in my home directory.

I hope this is helpful!

On Mon, Jun 15, 2015 at 9:20 PM, Ishrat Badami badami.ishrat@gmail.com
wrote:

> Thank you for the reply.
> 
> On Mon, Jun 15, 2015 at 8:14 PM, CNOT notifications@github.com wrote:
> 
> > No. Unfortunately I couldn't make Julia Studio work. I ended up using
> > julia in sublime text, which has a really nice interface but nevertheless
> > it's lacking some of the graphical aspects of Julia Studio.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/forio/julia-studio/issues/190#issuecomment-112159307
> > .
> 
> ## 
> 
> Ishrat Badami
> Ph.D. Student
> Computer Vision Group
> UMIC Research Centre
> RWTH Aachen University
> Room No 129
> Mies-van-der-Rohe Strasse 15
> D-52074 Aachen
> 
> ## Germany
> 
> Tel: +49 241 80 20775
> Fax: +49 241 80 22731
> 
> e-mail: badami@vision.rwth-aachen.de
> 
> ## URL: http://www.vision.rwth-aachen.de/people/ishrat-badami

## 

Ishrat Badami
Ph.D. Student
Computer Vision Group
UMIC Research Centre
RWTH Aachen University
Room No 129
Mies-van-der-Rohe Strasse 15
D-52074 Aachen

## Germany

Tel: +49 241 80 20775
Fax: +49 241 80 22731

e-mail: badami@vision.rwth-aachen.de

## URL: http://www.vision.rwth-aachen.de/people/ishrat-badami
